pkg/applier/inventory/inventory_converter.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inventory
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog/v2"
+	"kpt.dev/configsync/pkg/api/configsync"
+	"kpt.dev/configsync/pkg/core"
+	"kpt.dev/configsync/pkg/declared"
+	"kpt.dev/configsync/pkg/metadata"
+	"sigs.k8s.io/cli-utils/pkg/apis/actuation"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// UnstructuredInventoryConverter is an interface for converting ResourceGroup
+// objects to/from UnstructuredInventory objects.
+// This provides the translation logic to instantiate a UnstructuredClient which
+// works with ResourceGroup objects.
+type UnstructuredInventoryConverter interface {
+	// InventoryFromUnstructured converts a ResourceGroup object to UnstructuredInventory
+	InventoryFromUnstructured(fromObj *unstructured.Unstructured) (*inventory.UnstructuredInventory, error)
+	// InventoryToUnstructured converts a UnstructuredInventory to a ResourceGroup
+	// object, using the current ResourceGroup object as a base.
+	InventoryToUnstructured(fromObj *unstructured.Unstructured, toInv *inventory.UnstructuredInventory) (*unstructured.Unstructured, error)
+}
+
+// unstructuredInventoryConverter is used for converting ResourceGroup objects and persisting
+// Config Sync specific metadata.
+type unstructuredInventoryConverter struct {
+	syncKind      string
+	syncName      string
+	syncNamespace string
+	statusMode    string
+	// TODO: add source/commit hash and interface method once it's to be persisted on ResourceGroup
+}
+
+// NewInventoryConverter constructs a new UnstructuredInventoryConverter
+func NewInventoryConverter(scope declared.Scope, syncName, statusMode string) UnstructuredInventoryConverter {
+	if scope == declared.RootScope {
+		return &unstructuredInventoryConverter{
+			syncKind:      configsync.RootSyncKind,
+			syncName:      syncName,
+			syncNamespace: configsync.ControllerNamespace,
+			statusMode:    statusMode,
+		}
+	}
+	return &unstructuredInventoryConverter{
+		syncKind:      configsync.RepoSyncKind,
+		syncName:      syncName,
+		syncNamespace: string(scope),
```

Comment:
nit: scope.SyncNamespace()

Reply:
you can actually use the same for the RootScope too

Reply:
Done